specs/executionAPISpec.md

```diff
@@ -0,0 +1,530 @@
+* [ECAL-2] eth_createAccessList **MUST** use "latest" when the `defaultBlockParameter` is not specified.
+* [ECAL-3] eth_createAccessList **MUST** use the estimated gas when the `gas` is not specified.
+* [ECAL-4] eth_createAccessList **MUST** error with code -32000 when `gas` is too low.
+* [ECAL-5] eth_createAccessList **MUST** error with code -32000 when the `from` address does not have enough Ether to pay for the transaction.
+## <p id="estimateGas">eth_estimateGas</p>
+* [EEG-1] eth_estimateGas **MUST** return the estimated amount of gas the given `transaction` will take to execute.
+* [EEG-2] eth_estimateGas **MUST** error with code -32000 when the `from` address has insufficient Ether to execute the given `transaction`.
+* [EEG-3] eth_estimateGas **MUST NOT** check if the `from` account has sufficient funds when estimating contract deployment.
+* [EEG-4] eth_estimateGas **MUST** use 0x0000000000000000000000000000000000000000 for `from` when it is not specified.
+* [EEG-5] eth_estimateGas **MUST** error with code -32000 when estimating a contract creation that causes an error within the EVM.
+## <p id="signTransaction">eth_signTransaction</p>
+* [ESNT-1] eth_signTransaction **MUST** return both the signed encoded and signed unencoded format of the `transaction` given.
+* [ESNT-2] eth_signTransaction **MUST NOT** use given transaction `type` for transaction type, **MUST** determine transaction type from given transaction parameters.
+* [ESNT-3] eth_signTransaction **MUST** return encoded legacy transactions in the RLP format.
+* [ESNT-4] eth_signTransaction **MUST** return encoded [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) transactions as the transaction `type` concatenated with the transaction type's defined encoding method.
+* [ESNT-5] eth_signTransaction **MUST** use 0x0000000000000000000000000000000000000000 as default `from` address when `from` is null or not specified.
+* [ESNT-6] eth_signTransaction **MUST** error with code -32000 when the `from` address is locked.
+* [ESNT-7] eth_signTransaction **MUST** error with code -32000 when sending transaction using a `from` address that the client does not have the private key for.
+* [ESNT-8] eth_signTransaction **MUST** error with code -32000 when the `gas` is not specified.
+* [ESNT-9] eth_signTransaction **MUST** use 0x0 for `gas` when parameter is null.
+* [ESNT-10] eth_signTransaction **MUST** error with code -32000 when the `gasPrice` or `maxFeePerGas` and `maxPriorityFeePerGas` are not specified.
+* [ESNT-11] eth_signTransaction **MUST** error with code -32000 when `gasPrice` is used with `maxFeePerGas` and/or `maxPriorityFeePerGas`.
+* [ESNT-12] eth_signTransaction **MUST** use 0x0 for `gasPrice` or `maxFeePerGas` and `maxPriorityFeePerGas` when the parameter is null. 
```

Comment:
This will make the transaction unusable, which seems unlikely to be what the user wants.  Either the transaction signing should fail, or a gas price should be chosen by the signer.  I think it is incorrect to prescribe bad behavior in the specification.

Reply:
Same as below it needs to say `when the parameter is empty` not null. Using null errors which I can add to [ESNT-10]. I agree with you that the spec shouldn't prescribe bad behavior, but I didn't know if I had the authority to declare what I thought the API **should** do. I will write up a doc that lists all the situations like this to make it easier to see what needs to be discussed or changed on the clients.